int32  	id
string 	label
int32 	prev_id
string  prev_label
